File: Taller1_Python/Interfaz-Impresoras.py
"""
Este modulo contiene la interfaz de las impresoras
"""
import Impresoras
import threading
import random

delay_time = random.uniform(2.5,4.5)
limite_superior = -1;

try:
    limite_superior = int(input("Ingrese un numero > 0: "))
except:
    print("Valor no aceptado como numerico")

if limite_superior > 0:
    print(f"Se imprimiran numeros con un intervalo de espera de {delay_time} milisegundos")
    t1 = threading.Thread(target=Impresoras.impresora_pares, args = (limite_superior, delay_time,))
    t2 = threading.Thread(target=Impresoras.impresora_impares, args = (limite_superior, delay_time,))
    t1.start()
    t2.start()

File: src/interfaz/Interfaz_Impresoras.java
package interfaz;

import java.util.Scanner;

import modelo.Impresora_Impares;
import modelo.Impresora_Pares;

public class Interfaz_Impresoras {
	
	public void run() {
		Scanner scanner = new Scanner(System.in);
		System.out.println("Ingrese un numero > 0: ");
		
		int  limite_superior = -1;
		long delay_time = 0L + (long) (Math.random() * (4500L - 2500L));
		
		try {
			limite_superior = Integer.parseInt(scanner.nextLine());			
		} 
		catch(Exception e) {
			System.out.println("Valor no aceptado como numerico");
		}
		
		if(limite_superior > 0) {
		    System.out.println("Se imprimiran numeros con un intervalo de espera de " + delay_time + " milisegundos");
			Impresora_Pares		t1 = new Impresora_Pares(limite_superior, delay_time); 
			Impresora_Impares	t2 = new Impresora_Impares(limite_superior, delay_time); 
			t2.run();
			t1.run();
			
		}		
	}

	public static void main(String[] args) 
	{
		Interfaz_Impresoras interfaz = new Interfaz_Impresoras();
		interfaz.run();
	}
}
